Content.Server/MassMedia/Systems/NewsSystem.cs

```diff
@@ -0,0 +1,153 @@
+    {
+        if (!Resolve(deviceEnt, ref component))
+            return;
+
+        if (!TryComp<ActorComponent>(user, out var actor))
+            return;
+
+        if (_ui.TryToggleUi(deviceEnt, NewsReadUiKey.Key, actor.PlayerSession))
+        {
+
+        }
+
+        UpdateReadUi(deviceEnt, component);
+    }
+
+    public void ToggleUi(EntityUid user, EntityUid deviceEnt, NewsWriteComponent? component)
+    {
+        if (!Resolve(deviceEnt, ref component))
+            return;
+
+        if (!TryComp<ActorComponent>(user, out var actor))
+            return;
+
+        if (_ui.TryToggleUi(deviceEnt, NewsWriteUiKey.Key, actor.PlayerSession))
+        {
+
+        }
+
+        UpdateWriteUi(deviceEnt, component);
+    }
+
+    public void UpdateWriteUi(EntityUid uid, NewsWriteComponent component)
+    {
+        if (!_ui.TryGetUi(uid, NewsWriteUiKey.Key, out _))
+            return;
+
+        var state = new NewsWriteBoundUserInterfaceState(Articles.ToArray());
+        _ui.TrySetUiState(uid, NewsWriteUiKey.Key, state);
+    }
+
+    public void UpdateReadUi(EntityUid uid, NewsReadComponent component)
+    {
+        if (!_ui.TryGetUi(uid, NewsReadUiKey.Key, out _))
+            return;
+
+        if (component.ArticleNum < 0) NewsReadLeafArticle(component, false);
+
+        NewsReadBoundUserInterfaceState state;
+        if (Articles.Count != 0)
+            state = new NewsReadBoundUserInterfaceState(Articles[component.ArticleNum], component.ArticleNum + 1, Articles.Count);
+        else
+            state = new NewsReadBoundUserInterfaceState(new NewsArticle(), -1, -1);
+
+        _ui.TrySetUiState(uid, NewsReadUiKey.Key, state);
+    }
+
+    public void OnWriteUiMessage(EntityUid uid, NewsWriteComponent component, NewsWriteShareMessage msg)
+    {
+        Articles.Add(msg.Article);
+
+        UpdateWriteUi(uid, component);
+        TryNotify();
+    }
+
+    public void OnWriteUiMessage(EntityUid uid, NewsWriteComponent component, NewsWriteDeleteMessage msg)
+    {
+        component.Test = Articles.Count;
+
+        if (msg.ArticleNum < Articles.Count)
+        {
+            Articles.RemoveAt(msg.ArticleNum);
+        }
+
+        UpdateWriteUi(uid, component);
+    }
+
+    public void OnWriteUiMessage(EntityUid uid, NewsWriteComponent component, NewsWriteArticlesRequestMessage msg)
+    {
+        UpdateWriteUi(uid, component);
+    }
+
+    public void OnReadUiMessage(EntityUid uid, NewsReadComponent component, NewsReadLeafMessage msg)
+    {
+        component.Test++;
+
+        NewsReadLeafArticle(component, msg.IsNext);
+
+        UpdateReadUi(uid, component);
+    }
+
+    private void NewsReadLeafArticle(NewsReadComponent component, bool isNext)
+    {
+        if (isNext)
+            component.ArticleNum++;
+        else
+            component.ArticleNum--;
+
+        if (component.ArticleNum >= Articles.Count) component.ArticleNum = 0;
+        if (component.ArticleNum < 0) component.ArticleNum = Articles.Count - 1;
```

Comment:
`component.ArticleNum = component.ArticleNum + Articles.Count % Articles.Count`
to replace both ifs

Reply:
It doesn't work, although most likely I'm an idiot...

Reply:
Can you explain what behavior doesn't work?

Reply:
If I understand this equation correctly, it should limit the value instead of ifs, but I don't understand how it can work like this